### PR TITLE
libtransmission: init at 3.00

### PR DESCRIPTION
**Pull request by onny**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by prusnak** [comment]
Wouldn't it be better to just add `installLib ? false` option to `pkgs/applications/networking/p2p/transmission/default.nix`:

``` patch
--- a/pkgs/applications/networking/p2p/transmission/default.nix
+++ b/pkgs/applications/networking/p2p/transmission/default.nix
@@ -20,6 +20,7 @@
 , enableSystemd ? stdenv.isLinux
 , enableDaemon ? true
 , enableCli ? true
+, installLib ? false
 }:
 
 let
@@ -47,6 +48,7 @@ in stdenv.mkDerivation {
       "-DENABLE_QT=${mkFlag enableQt}"
       "-DENABLE_DAEMON=${mkFlag enableDaemon}"
       "-DENABLE_CLI=${mkFlag enableCli}"
+      "-DINSTALL_LIB=${mkFlag installLib}"
     ];
 
   nativeBuildInputs = [
```

and then use the following in `pkgs/top-level/all-packages.nix`:

``` patch
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -26099,6 +26099,7 @@ in
   transcode = callPackage ../applications/audio/transcode { };
 
   transmission = callPackage ../applications/networking/p2p/transmission { };
+  libtransmission = transmission.override { installLib = true; enableDaemon = false; enableCli = false; };
   transmission-gtk = transmission.override { enableGTK3 = true; };
   transmission-qt = transmission.override { enableQt = true; };
 
```

This would remove lots of unnecessary duplication.

**Comment by veprbl** [comment]
@GrahamcOfBorg eval

File: pkgs/development/libraries/libtransmission/default.nix [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        * Full encryption, DHT, and PEX support
+    '';
+    homepage = "http://www.transmissionbt.com/";
+    license = lib.licenses.gpl2; # parts are under MIT
+    platforms = lib.platforms.unix;
+    maintainers = with maintainers; [ onny ];
+  };
+
+}
+
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libtransmission/default.nix:82]:
```suggestion
  };
}
```

File: pkgs/development/libraries/libtransmission/default.nix [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  ]
+  ++ lib.optionals enableSystemd [ systemd ]
+  ++ lib.optionals stdenv.isLinux [ inotify-tools ]
+  ;
+
+  NIX_LDFLAGS = lib.optionalString stdenv.isDarwin "-framework CoreFoundation";
+
+  installFlags = [ "-C" "libtransmission" ];
+
+  meta = {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libtransmission/default.nix:62]:
```suggestion
  meta = with lib; {
```

File: pkgs/development/libraries/libtransmission/default.nix [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        * Uses fewer resources than other clients
+        * Native Mac, GTK and Qt GUI clients
+        * Daemon ideal for servers, embedded systems, and headless use
+        * All these can be remote controlled by Web and Terminal clients
+        * Bluetack (PeerGuardian) blocklists with automatic updates
+        * Full encryption, DHT, and PEX support
+    '';
+    homepage = "http://www.transmissionbt.com/";
+    license = lib.licenses.gpl2; # parts are under MIT
+    platforms = lib.platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libtransmission/default.nix:77]:
```suggestion
    license = with licenses; [ gpl2 mit ];
    platforms = platforms.unix;
```

Also gpl2Only or gpl2Plus?

File: pkgs/development/libraries/libtransmission/default.nix [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  buildInputs = [
+    openssl
+    curl
+    libevent
+    zlib
+    pcre
+  ]
+  ++ lib.optionals enableSystemd [ systemd ]
+  ++ lib.optionals stdenv.isLinux [ inotify-tools ]
+  ;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libtransmission/default.nix:56]:
```suggestion
  ] ++ lib.optionals enableSystemd [ systemd ]
  ++ lib.optionals stdenv.isLinux [ inotify-tools ];
```

File: pkgs/development/libraries/libtransmission/default.nix [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    in
+    [
+      "-DENABLE_MAC=OFF" # requires xcodebuild
+      "-DENABLE_DAEMON=OFF"
+      "-DENABLE_CLI=OFF"
+      "-DINSTALL_LIB=ON"
+    ];
+
+  nativeBuildInputs = [
+    pkgconfig
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libtransmission/default.nix:43]:
```suggestion
    pkg-config
```

File: pkgs/development/libraries/libtransmission/default.nix [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    owner = "transmission";
+    repo = "transmission";
+    rev = version;
+    sha256 = "0ccg0km54f700x9p0jsnncnwvfnxfnxf7kcm7pcx1cj0vw78924z";
+    fetchSubmodules = true;
+  };
+
+  cmakeFlags =
+    let
+      mkFlag = opt: if opt then "ON" else "OFF";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libtransmission/default.nix:33]: Why is there no lib function for this?

File: pkgs/development/libraries/libtransmission/default.nix [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+, pcre
+, enableSystemd ? stdenv.isLinux
+}:
+
+let
+  version = "3.00";
+
+in stdenv.mkDerivation {
+  pname = "libtransmission";
+  inherit version;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libtransmission/default.nix:21]:
```suggestion
stdenv.mkDerivation rec {
  pname = "libtransmission";
  version = "3.00";
```

File: pkgs/development/libraries/libtransmission/default.nix [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+  src = fetchFromGitHub {
+    owner = "transmission";
+    repo = "transmission";
+    rev = version;
+    sha256 = "0ccg0km54f700x9p0jsnncnwvfnxfnxf7kcm7pcx1cj0vw78924z";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libtransmission/default.nix:28]: Ideally those should be devendored but they are all from transmission, too so it is fine I think.

File: pkgs/development/libraries/libtransmission/default.nix [hunk 9]
```diff
@@ -0,0 +1,82 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, cmake
+, pkgconfig
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libtransmission/default.nix:5]:
```suggestion
, pkg-config
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/libtransmission/default.nix [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+        * Bluetack (PeerGuardian) blocklists with automatic updates
+        * Full encryption, DHT, and PEX support
+    '';
+    homepage = "http://www.transmissionbt.com/";
+    license = with licenses; [ gpl2Only mit ];
+    platforms = lib.platforms.unix;
+    maintainers = with maintainers; [ onny ];
+  };
+}
+
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libtransmission/default.nix:76]: One final new line to much.

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/development/libraries/libtransmission/default.nix [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+, curl
+, libevent
+, inotify-tools
+, systemd
+, zlib
+, pcre
+, enableSystemd ? stdenv.isLinux
+}:
+
+stdenv.mkDerivation {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libtransmission/default.nix:16]:
```suggestion
stdenv.mkDerivation rec {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by ajs124** [comment]
I'd have to agree with what @prusnak said. Duplicating this seems kind of weird, when we already have transmission and could just add those 3 lines of code.

**Comment by prusnak** [comment]
@onny will you rework the PR according to my suggestion, please?

**Pull request merged**